### ISSUE TITLE
Weirdness with inline figure DPI settings in Jupyter Notebook

### ISSUE THREAD

**Issue by kiwirob** [matplotlib/matplotlib#9217]
I find the default inline plotting DPI (72) of figures in Jupyter too small and would like to plot inline figures at 100 by default. However, the (i) import of pyplot and (ii) use of %matplotlib inline both seem to override the figure.dpi setting of 100 I have put in my .matplotlibrc file. I can address this issue by using mpl.rcParams['figure.dpi'] = 100 after importing pyplot and running %matplotlib inline, but think it would be better if this was not required every time.

I'm not sure if this is an issue with matplotlib, Jupyter Notebook and/or ipykernel. 

Any assistance appreciated.

Issue 1: Import of pyplot overrides figure.dpi in .matplotlibrc
![001](https://user-images.githubusercontent.com/4974928/30753798-4fee1c46-9fb8-11e7-8827-934bcf4bfdf6.PNG)

Issue 2: Use of %matplotlib inline magic overrides figure.dpi in .matplotlibrc
![002](https://user-images.githubusercontent.com/4974928/30753808-5ec50c02-9fb8-11e7-9be5-e2f73469bbfc.PNG)

**Matplotlib version**

  * Operating System: Win32 
  * Matplotlib Version: 2.0.2
  * Python Version: 3.6
  * Jupyter Version : Notebook 5.0, Jupyter core 4.3
  * iPython: 5.3, ipykernel: 4.6.1

**Comment by jklymak** [comment]
Any reason to not use ` %matplotlib notebook`?

**Comment by tacaswell** [comment]
attn @Carreau

**Comment by Carreau** [comment]
Yes, the inline backend has different rcParam (historically) we should likely revisit those now that there is matplotlib 2;  Screen resolution has also evolved quite a bit since we added that. 

I guess the "Correct" way to do it would be to have a notebook style, and activate that style when doing inline. 

```
rc = Dict({'figure.figsize': (6.0,4.0),
        # play nicely with white background in the Qt and notebook frontend
        'figure.facecolor': (1,1,1,0),
        'figure.edgecolor': (1,1,1,0),
        # 12pt labels get cutoff on 6x4 logplots, so use 10pt.
        'font.size': 10,
        # 72 dpi matches SVG/qtconsole
        # this only affects PNG export, as SVG has no dpi setting
        'figure.dpi': 72,
        # 10pt still needs a little more room on the xlabel:
        'figure.subplot.bottom' : .125
        },
        help="""Subset of matplotlib rcParams that should be different for the
        inline backend."""
    ).tag(config=True)
```

I'm pretty sure there is already an issue with that, but I could not find it, so I opened https://github.com/ipython/ipykernel/issues/267

**Comment by kiwirob** [comment]
Thanks for looking into this. I can see why the IPython inline backend rc param over-ride is causing the second part of my original issue. Per [this SO thread](https://stackoverflow.com/questions/35939989/matplotlib-configuration-for-inline-backend-in-jupyter-notebook), this can be addressed by setting `c.InlineBackend.rc` to an empty dict in `~/.ipython/profile_default/ipython_kernel_config.py`.

However, re the first part of my original issue, is there any reason that the `matplotlib.pyplot` import call alone (i.e. without `%matplotlib inline`) in Jupyter Notebook should be changing matplotlib `figure.dpi` rc param? i.e.

![](https://user-images.githubusercontent.com/4974928/30753798-4fee1c46-9fb8-11e7-8827-934bcf4bfdf6.PNG)

**Comment by michaelaye** [comment]
I also just noticed that this difference in DPI apparently has grown since MPL 2.x.
https://gist.github.com/ccac0541d040626b1c57e64d874fabe9

**Comment by darthoctopus** [comment]
I'm also affected by the first part of the problem after updating matplotlib and Jupyter today (happened after updating Jupyter), and would like to know how to disable this.

**Comment by vnoel** [comment]
I too find too small the matplotlib figures created in jupyter.
Unfortunately, @Carreau I do not understand your advice... What is a notebook style? How can I  activate that style when I do inline? Where should I copy/paste your snippet? Thanks!

**Comment by talhum** [comment]
Having the `%matplotlib inline` and `mpl.rcParams['figure.dpi'] = 150` in the same cell does not work as expected: Even if the magic command (`%matplotlib inline`) is placed before the assignment line (`mpl.rcParams['figure.dpi'] = 150`), it is called last and overwrites `figure.dpi`.

The work-around solution is to keep the two commands in two separate cells and run the cell with `%matplotlib inline` before that of `mpl.rcParams['figure.dpi'] = 150`.

**Comment by soerendip** [comment]
Could the magic function just no override the rc settings.

**Comment by ariel-co** [comment]
I'm not sure if this is the right place to ask, but I wonder why now the override is only apparent _after_ the cell where pyplot is imported has run:
![image](https://user-images.githubusercontent.com/8219794/71328713-1cace580-24f2-11ea-9a56-1817d38368ea.png)

**Comment by dlukes** [comment]
Just FYI, https://github.com/ipython/ipykernel/issues/267 is now resolved with `matplotlib-inline` 0.1.6, so there should be no more weirdness around this:

![image](https://user-images.githubusercontent.com/2734517/188473356-534f5c7a-6341-423c-9b45-a690c6f674f4.png)

**Comment by christianbrodbeck** [comment]
Now what happens for me is that the parameter changes after creating the first plot: 

<img width="461" alt="Screenshot 2023-06-26 at 2 05 50 PM" src="https://github.com/matplotlib/matplotlib/assets/145771/dbaf9191-9662-4c6c-b3b8-0922b59d11ee">

**Comment by dlukes** [comment]
What about `!pip show matplotlib-inline`? If it's less than 0.1.6, you need to upgrade it.

**Comment by timhoffm** [comment]
It may be that this is the intended behavior. We nowadays defer backend selection until the first plot. If the activated backend manipulates rcParams, this is obviously deferred as well. 

You can explicitly select the backend through https://matplotlib.org/stable/api/matplotlib_configuration_api.html#matplotlib.use. `%matplotlib inline`

**Comment by christianbrodbeck** [comment]
Updating `matplotlib-inline` fixed it, thanks @dlukes ! I did not realize the backend is a separate package.
